gremlin-test/src/main/java/org/apache/tinkerpop/gremlin/AbstractGraphProvider.java

```diff
@@ -151,5 +151,7 @@ protected void readIntoGraph(final Graph graph, final String path) throws IOExce
         final String dataFile = TestHelper.generateTempFileFromResource(graph.getClass(),
                 GryoResourceAccess.class, path.substring(path.lastIndexOf(File.separator) + 1), "", false).getAbsolutePath();
         graph.traversal().io(dataFile).read().iterate();
+        if (graph.features().graph().supportsTransactions())
```

Comment:
i'm pretty sure we left this code out purposefully and meant for the graph to handle commit after read on its own. unless there's a good reason to change this, i'd say it would be better to move this code to TinkerGraph's test implementation and just add to the javadoc to note that implementers should handle commit on their own.

Reply:
addressed in https://github.com/apache/tinkerpop/pull/2087/commits/595d8273753c1d4bf3ed072cf9afd5215dcc0420